--- !ditz.rubyforge.org,2008-03-06/issue 
title: Timeout::timeout() is troublesome
desc: Timeout::timeout() sometimes kills a thread outside of it's catch clause which causes the whole reaper to die.
type: :bugfix
component: Reaper
release: 
reporter: David Triendl <david@triendl.name>
status: :closed
disposition: :fixed
creation_time: 2010-12-02 13:59:31.092731 Z
references: []

id: dfa4bba4bcce52b6ffd5c3af13d7ecdd93194d5d
log_events: 
- - 2010-12-02 13:59:31.632226 Z
  - David Triendl <david@triendl.name>
  - created
  - ""
- - 2010-12-02 20:49:22.130999 Z
  - David Triendl <david@triendl.name>
  - changed status from unstarted to in_progress
  - Trying to avoid Thread.raise.
- - 2010-12-02 21:06:28.607311 Z
  - David Triendl <david@triendl.name>
  - closed with disposition fixed
  - Using Thread.kill instead of Timeout's Thread.raise now, which avoid the proublem with uncaught exceptions etc.
